server/store/user_store.go

```diff
@@ -85,3 +135,33 @@ func (s *pluginStore) DeleteUser(mattermostUserID string) error {
 	}
 	return nil
 }
+
+func (index UserIndex) ByMattermostID() map[string]*UserShort {
+	result := map[string]*UserShort{}
+
+	for _, u := range index {
+		result[u.MattermostUserID] = u
+	}
+
+	return result
+}
+
+func (index UserIndex) ByRemoteID() map[string]*UserShort {
+	result := map[string]*UserShort{}
+
+	for _, u := range index {
+		result[u.RemoteID] = u
+	}
+
+	return result
+}
+
+func (index UserIndex) ByEmail() map[string]*UserShort {
```

Comment:
Instead of deserializing into a map, I have chosen to have these helper methods that create maps for the different use cases. The `ByEmail` method is the only one used at the moment. The email is unique per each Microsoft account.